### PR TITLE
Select component test

### LINKED ISSUE #495: Select Component Test
/components/Select

### PR DESCRIPTION
**Pull request by SafeDuck**
image

### CODE REVIEW

File: cypress/component/Select.cy.jsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
+import Select from "@/components/Select";
+describe("Select", () => {
+  const options = ["Option 1", "Option 2", "Option 3"];
+  const field = "field";
+  const placeholder = "Select an option";
+  const title = "Select an option";
+
+  it("Renders", () => {
+    const setUserStub = cy.stub();
```
Comment (shahdivyank) [inline cypress/component/Select.cy.jsx]: make this a hook that u can use

File: cypress/component/Select.cy.jsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+    cy.get('[data-cy="toggle"]').should("exist");
+    cy.get('[data-cy="menu"]').should("not.exist");
+  });
+
+  it("Show dropdown menu", () => {
+    const setUserStub = cy.stub();
+
+    const Parent = () => {
+      const user = { [field]: "" };
```
Comment (shahdivyank) [inline cypress/component/Select.cy.jsx]: use react hooks and u will be able to test the actual data manipulation

File: cypress/component/Select.cy.jsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+describe("Select", () => {
+  const options = ["Option A", "Option B", "Option C"];
+  const field = "field";
+  const placeholder = "Select an option";
+  const title = "Title";
+
+  it("Renders", () => {
+    const Parent = () => {
+      const [user, setUser] = useState({ [field]: "" });
```
Comment (shahdivyank) [inline cypress/component/Select.cy.jsx]: dont make the field a variable, use a string, also use a better attribute, something more relevant like "name"

File: cypress/component/Select.cy.jsx [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+      );
+    };
+
+    cy.mount(<Parent />);
+
+    cy.get('[data-cy="toggle"]').should("exist");
+    cy.get('[data-cy="menu"]').should("not.exist");
+  });
+
+  it("Show dropdown menu", () => {
```
Comment (shahdivyank) [inline cypress/component/Select.cy.jsx]:
```suggestion
  it("Show Dropdown", () => {
```

File: cypress/component/Select.cy.jsx [hunk 5]
```diff
@@ -0,0 +1,52 @@
+import Select from "@/components/Select";
```
Comment (shahdivyank) [inline cypress/component/Select.cy.jsx:1]: add a test that selects an option and then make sure that update actually happened

Comment (SafeDuck) [reply cypress/component/Select.cy.jsx:1]: isn't that already in the e2e tests for sort?

Comment (shahdivyank) [reply cypress/component/Select.cy.jsx:1]: comp test should make sure that the entire workflow for a comp works. testing the select values and stuff is comp test cause it is handled internally by the comp and not by a parent comp

File: src/components/Select.jsx [hunk 6]
```diff
@@ -0,0 +1,58 @@
+import Dropdown from "react-bootstrap/Dropdown";
+
+const Select = ({
```
Comment (shahdivyank) [inline src/components/Select.jsx]: uh there is a whole component being added

Comment (SafeDuck) [reply src/components/Select.jsx]: ah sheet ma bad, weird merging. Menthy suggested to delete it from my branch

**Review by shahdivyank — APPROVED** [review]

**Pull request merged**